configure.ac

```diff
@@ -2423,10 +2418,10 @@ else
       # Set Scilab startup options depending on version
       AC_MSG_CHECKING(for Scilab startup options)
       SCILABOPT="-nwni -nb"
-      if test $SCILAB_VERSION_NO_DOTS -ge 540; then
+      if test $(echo -e '$SCILAB_VERSION\n5.3.3' |sort -n |head -n1) != 5.3.3; then
```

Comment:
This trick doesn't actually work correctly - e.g. 5.3.10 compare as less than 5.3.3 (`-n` sorts the bit that looks like a floating point number numerically, but the second `.` and everything after appears to just get resolved with a string comparison):

```
$ printf '%s\n' 5.3.3 5.3.4 5.3.10 | sort -n
5.3.10
5.3.3
5.3.4
```

The existing "NO_DOTS" comparison is also broken though.

Reply:
That's not really suitable as there are platforms using other `sort` implementations.

Did you switch from the `SCILAB_VERSION_NO_DOTS` approach because it fails to handle multi-digit version components correctly?

If so, zero padding them to 3 each isn't hard:

```
SCILAB_VERSION_NO_DOTS=`printf '%03d%03d%03d' "$SCILAB_MAJOR_VERSION" "$SCILAB_MINOR_VERSION" "$SCILAB_MAINTENANCE_VERSION"`
```

E.g. 5.3.10 would then give `005003010`

Reply:
Oh I see the motivation was that the major version is now `2003`.  The above would work since `2003.0.0` gives (`2003000000`) but versions above 2147.483.648 potentially overflow a 32-bit int.

If that's a concern, using `%d%02d%02d` as the format would work if `SCILAB_MINOR_VERSION` and `$SCILAB_MAINTENANCE_VERSION` never go above 99.

Otherwise we could make the format `%04d%03d%03d` and use string comparisons like:

```
SCILAB_VERSION_NO_DOTS=`printf '%04d%03d%03d' "$SCILAB_MAJOR_VERSION" "$SCILAB_MINOR_VERSION" "$SCILAB_MAINTENANCE_VERSION"`
if test "$SCILAB_VERSION_NO_DOTS" '>' 0005003002; then 
```

Slightly awkward is you need to remember to quote `>` (otherwise it's interpreted as file redirection) and also `>=` doesn't work so you need to use `>` and compare against a version one lower.

Alternatively `expr` supports `>=` which does a string compare if the arguments can't be converted to integers:

```
if expr X"$SCILAB_VERSION_NO_DOTS" '>=' X0005003003 >/dev/null; then
```

Reply:
Thanks for your help, I used the `%04d%02d%02d` format to normalize the version and detect the same versions as before using the same `test` expressions.
```
SCILAB_VERSION_NO_DOTS=$(printf '%04d%03d%03d' "$SCILAB_MAJOR_VERSION" "$SCILAB_MINOR_VERSION" "$SCILAB_MAINTENANCE_VERSION")
```